File: QuizApp/app/src/main/java/com/example/quizapp/MainActivity.java
package com.example.quizapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private TextView mQuestgionText;
    private Button mTrueButton,mFalseButton, mNextButton;
    private int mCurrentIndex = 0;

    private TrueFalse[] mQuestions = new TrueFalse[]{
            new TrueFalse(R.string.question_one,true),
            new TrueFalse(R.string.question_two,true),
            new TrueFalse(R.string.question_three,false),
            new TrueFalse(R.string.question_four,true),
            new TrueFalse(R.string.question_five,true),
            new TrueFalse(R.string.question_six,false),
            new TrueFalse(R.string.question_seven,true),
            new TrueFalse(R.string.question_eight,false)
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mQuestgionText = findViewById(R.id.question_text_view);
        final int Question = mQuestions[mCurrentIndex].getmQuestion();
        mQuestgionText.setText(Question);

        mTrueButton = findViewById(R.id.true_button);
        mTrueButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                checkAnswer(true);
            }
        });

        mFalseButton = findViewById(R.id.false_button);
        mFalseButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                checkAnswer(false);
            }
        });

        mNextButton = findViewById(R.id.next_button);
        mNextButton.setOnClickListener(new View.OnClickListener(){
            public void onClick(View view){
                mCurrentIndex = (mCurrentIndex+1) % mQuestgionText.length();
                int question = mQuestions[mCurrentIndex].getmQuestion();
                mQuestgionText.setText(question);
            }
        });


    }
    private void  checkAnswer(boolean userPerssed){
        boolean answer = mQuestions[mCurrentIndex].ismTrueQuestion();
        if(userPerssed == answer){
            Toast.makeText(MainActivity.this, R.string.true_message, Toast.LENGTH_LONG).show();
        }else{
            Toast.makeText(MainActivity.this, R.string.false_message,Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/example/app/MainActivity.kt
package com.example.app

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_linstener)

        //        화면을 그려주는 함수

        Log.d("life_cycle","onCreate")
    } //오버라이드
    override fun onStart() {
        super.onStart()
        Log.d("life_cycle","onCreate")
    }

    override fun onResume() {
        super.onResume()
        Log.d("life_cycle","onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d("life_cycle","onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d("life_cycle","onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("life_cycle","onDestroy")
    }


}